### PR TITLE
Add GDAL 3.0.0

### PR DESCRIPTION
**Pull request by adamjstewart**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gdal/package.py [hunk 1]
```diff
@@ -123,7 +125,10 @@ class Gdal(AutotoolsPackage):
     depends_on('opencl', when='+opencl')
     depends_on('poppler', when='+poppler')
     depends_on('poppler@:0.63', when='@:2.3.0 +poppler')
-    depends_on('proj', when='+proj @2.3:')
+    depends_on('proj@:4', when='+proj @2.3.0:2.3.999')
+    depends_on('proj@:5', when='+proj @2.4.0:2.4.999')
+    depends_on('proj@:6', when='+proj @2.5:2.999')
+    depends_on('proj@6:', when='+proj @3:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gdal/package.py]: Version requirements come from https://github.com/OSGeo/PROJ/wiki/proj.h-adoption-status

File: var/spack/repos/builtin/packages/gdal/package.py [hunk 2]
```diff
@@ -172,10 +199,6 @@ def configure_args(self):
 
             if '+proj' in spec:
                 args.append('--with-proj={0}'.format(spec['proj'].prefix))
-                if spec.satisfies('^proj@5.0:5.999'):
-                    args.append('--with-proj5-api=yes')
-                else:
-                    args.append('--with-proj5-api=no')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/gdal/package.py]: Why was this removed?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/gdal/package.py]: I'm not sure if it's actually necessary, and the flag no longer exists in newer versions.
